optimum/exporters/openvino/__main__.py

```diff
@@ -351,16 +351,22 @@ class StoreAttr(object):
                 GPTQQuantizer.post_init_model = post_init_model
     elif library_name == "diffusers" and is_openvino_version(">=", "2024.6"):
         _loading_kwargs = {} if variant is None else {"variant": variant}
-        dtype = deduce_diffusers_dtype(
-            model_name_or_path,
-            revision=revision,
-            cache_dir=cache_dir,
-            token=token,
-            local_files_only=local_files_only,
-            force_download=force_download,
-            trust_remote_code=trust_remote_code,
-            **_loading_kwargs,
-        )
+        dtype = loading_kwargs.pop("torch_dtype", None)
+        if isinstance(dtype, str):
+            dtype = None if dtype == "auto" else getattr(torch, dtype)
+        if ov_config is not None and ov_config.dtype in {"fp16", "fp32"}:
+            dtype = torch.float16 if ov_config.dtype == "fp16" else torch.float32
```

Comment:
what do you think about moving this before the library check as this code should be the same for transformers and diffusers models (expect for `deduce_diffusers_dtype` when dtype not inferred)
also currently it seems that for transformers models `torch_dtype` is prioritized when defined and if not then we check for the `ov_config` (only modified if  `torch_dtype` is not `None`), here it looks like the other way around (`dtype` will be overwritten)
https://github.com/huggingface/optimum-intel/blob/6cceb30aaf1a0126d054d726842ed4945c0227ba/optimum/exporters/openvino/__main__.py#L306-L319

Reply:
I though about that but the problem that auto treated differently for diffusers and transformers (in transformers, if auto specified, specified in config dtype will be used, in diffusers there is no similar behaviour, only explicit dtype can be used and there is no config file that contains this info, that kis why we use deduce_diffusers_dtype to resolve auto dtype)

Reply:
I moved common part (before resolving auto before library-specific code and also add ov_config precision check only for case when precision is not specified. Please check one more time

Reply:
nice, that works for me!